mmtrack/datasets/sot_dataset.py

```diff
@@ -0,0 +1,227 @@
+                    'start_frame': the starting frame number contained in the
+                                    image name
+                    'end_frame': the ending frame number contained in the image
+                                    name
+                    'num_digit': the number of digits in the image name
+                },
+                ...
+            ]
+        '''
+        self.data_infos = self.load_data_infos(split=self.split)
+        self.num_frames_per_video = []
+        for video_ind in range(len(self.data_infos)):
+            self.num_frames_per_video.append(self.get_len_per_video(video_ind))
+
+    def __getitem__(self, ind):
+        if self.test_mode:
+            assert isinstance(ind, tuple)
+            return self.prepare_test_img(ind[0], ind[1])
+        else:
+            return self.prepare_train_img(ind)
+
+    @abstractmethod
+    def load_data_infos(self, split='train'):
+        pass
+
+    def get_img_names_from_video(self, video_ind):
+        img_names = []
+        start_frame, end_frame = self.data_infos[video_ind][
+            'start_frame'], self.data_infos[video_ind]['end_frame']
+        num_digit = self.data_infos[video_ind]['num_digit']
+        for frame_id in range(start_frame, end_frame + 1):
+            img_names.append(
+                osp.join(
+                    self.data_infos[video_ind]['video_path'],
+                    '{frame_id:0{num_digit}}.jpg'.format(
+                        frame_id=frame_id, num_digit=num_digit)))
+        return img_names
+
+    def get_bboxes_from_video(self, video_ind):
+        bbox_path = osp.join(self.img_prefix,
+                             self.data_infos[video_ind]['ann_path'])
+        bboxes = np.loadtxt(bbox_path, dtype=float, delimiter=',')
+        if len(bboxes.shape) == 1:
+            bboxes = np.expand_dims(bboxes, axis=0)
+
+        end_frame, start_frame = self.data_infos[video_ind][
+            'end_frame'], self.data_infos[video_ind]['start_frame']
+
+        if not self.test_mode:
+            assert len(bboxes) == (
+                end_frame - start_frame +
+                1), f'{len(bboxes)}, {end_frame}, {start_frame}'
+        return bboxes
+
+    def get_len_per_video(self, video_ind):
+        return self.data_infos[video_ind]['end_frame'] - self.data_infos[
+            video_ind]['start_frame'] + 1
+
+    def get_visibility_from_video(self, video_ind):
+        visible = np.array([True] * self.get_len_per_video(video_ind))
+        return dict(visible=visible)
+
+    def get_masks_from_video(self, video_ind):
+        pass
+
+    def get_ann_infos_from_video(self, video_ind):
+        bboxes = self.get_bboxes_from_video(video_ind)
+        visible_info = self.get_visibility_from_video(video_ind)
+        bboxes_isvalid = (bboxes[:, 2] > self.bbox_min_size) & (
+            bboxes[:, 3] > self.bbox_min_size)
+        visible_info['visible'] = visible_info['visible'] & bboxes_isvalid
```

Comment:
I remember that we should not merge `visible` and `bbox_isvalid` in here

Reply:
It follows the `pytracking`. Generally, we think the object with 'visible'=True can be sampled directly, so it's annotation must valid and we merge `visible` and `bbox_isvalid`.